styx-api-service/src/main/java/com/spotify/styx/api/StatusResource.java

```diff
@@ -139,6 +144,29 @@ private static String arg(String name, RequestContext rc) {
     return Response.forPayload(RunStateDataPayload.create(runStates));
   }
 
+  private Map<WorkflowInstance, RunState> getActiveStates(String componentsStr) throws IOException {
+    final List<String> components = Arrays.asList(componentsStr.split(","));
+    final ImmutableMap.Builder<WorkflowInstance, RunState> mapBuilder = ImmutableMap.builder();
+
+    final List<Optional<IOException>> exceptions = components.parallelStream().map( componentId -> {
+      Optional<IOException> exception = Optional.empty();
+      try{
+        final Map<WorkflowInstance, RunState> stateMap = storage.readActiveStates(componentId);
+        for (WorkflowInstance instance : stateMap.keySet()) {
+          mapBuilder.put(instance, stateMap.get(instance));
+        }
+      } catch (IOException e) {
+        exception = Optional.of(e);
```

Comment:
Instead of using a variable, we can just return `Optional.of(e)` here and return `Optional.empty()` below